{
  "id": "PYSEC-2020-34",
  "modified": "2023-12-06T01:00:24.135957Z",
  "published": "2020-09-01T13:15:00Z",
  "aliases": [
    "BIT-django-2020-24584",
    "CVE-2020-24584",
    "GHSA-fr28-569j-53c4"
  ],
  "details": "An issue was discovered in Django 2.2 before 2.2.16, 3.0 before 3.0.10, and 3.1 before 3.1.1 (when Python 3.7+ is used). The intermediate-level directories of the filesystem cache had the system's standard umask rather than 0o077.",
  "affected": [
    {
      "package": {
        "ecosystem": "PyPI",
        "name": "django",
        "purl": "pkg:pypi/django"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "2.2"
            },
            {
              "fixed": "2.2.16"
            },
            {
              "introduced": "3.0"
            },
            {
              "fixed": "3.0.10"
            },
            {
              "introduced": "3.1"
            },
            {
              "fixed": "3.1.1"
            }
          ]
        }
      ],
      "versions": [
        "2.2",
        "2.2.1",
        "2.2.2",
        "2.2.3",
        "2.2.4",
        "2.2.5",
        "2.2.6",
        "2.2.7",
        "2.2.8",
        "2.2.9",
        "2.2.10",
        "2.2.11",
        "2.2.12",
        "2.2.13",
        "2.2.14",
        "2.2.15",
        "3.0",
        "3.0.1",
        "3.0.2",
        "3.0.3",
        "3.0.4",
        "3.0.5",
        "3.0.6",
        "3.0.7",
        "3.0.8",
        "3.0.9",
        "3.1"
      ],
      "database_specific": {
        "source": "https://github.com/pypa/advisory-database/blob/main/vulns/django/PYSEC-2020-34.yaml"
      }
    }
  ],
  "references": [
    {
      "type": "WEB",
      "url": "https://groups.google.com/forum/#!topic/django-announce/Gdqn58RqIDM"
    },
    {
      "type": "WEB",
      "url": "https://www.openwall.com/lists/oss-security/2020/09/01/2"
    },
    {
      "type": "WEB",
      "url": "https://groups.google.com/forum/#!topic/django-announce/zFCMdgUnutU"
    },
    {
      "type": "ARTICLE",
      "url": "https://www.djangoproject.com/weblog/2020/sep/01/security-releases/"
    },
    {
      "type": "WEB",
      "url": "https://docs.djangoproject.com/en/dev/releases/security/"
    },
    {
      "type": "WEB",
      "url": "https://usn.ubuntu.com/4479-1/"
    },
    {
      "type": "WEB",
      "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/F2ZHO3GZCJMP3DDTXCNVFV6ED3W64NAU/"
    },
    {
      "type": "WEB",
      "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ZCRPQCBTV3RZHKVZ6K6QOAANPRZQD3GI/"
    },
    {
      "type": "ADVISORY",
      "url": "https://security.netapp.com/advisory/ntap-20200918-0004/"
    },
    {
      "type": "WEB",
      "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OLGFFLMF3X6USMJD7V5F5P4K2WVUTO3T/"
    },
    {
      "type": "WEB",
      "url": "https://www.oracle.com/security-alerts/cpujan2021.html"
    },
    {
      "type": "ADVISORY",
      "url": "https://github.com/advisories/GHSA-fr28-569j-53c4"
    }
  ]
}